### DISCUSSION TITLE
Constant Pause and Parking

### DISCUSSION THREAD

**Discussion by cmacnichol** [mriscoc/Ender3V2S1#907] [Q&A] [answered]
Ender 3 V2 - Stock Screen - BLTouch, No End Stop Switch

BTT SKR Mini E3 V3

Ender3 V2/S1 Professional Firmware 20230312

I am really clueless at this point and could use some help and\or guidance. I just upgraded by mainboard after the previous one stopped working and for the life of me I cannot get it to work correctly. Any print I start, from Octoprint or Manually, immediately pauses and parks.

The Endstop diagnostic screen shows that Z_Min is staying in the TRIGGERED state instead of OPEN. I thought this meant the BLTouch was wired wrong, but Z-Homing and Leveling\Mesh Creation all work without an issue which implies the BLTouch is working normally. Further, I have no other Z Endstop plugged in to trigger it. The BLTouch is currently using the 5Pin JST connector on the board.

Any Thoughts on what this could be or anything else that I can check?

**Comment by mriscoc** [comment]
> Any print I start, from Octoprint or Manually, immediately pauses and parks.  

Check your filament runout sensor (filament settings menu).

> The Endstop diagnostic screen shows that Z_Min is staying in the TRIGGERED state instead of OPEN. I thought this meant the BLTouch was wired wrong.  

No, BLTouch has his own connector, Z_MIN refers to the Z_MIN end stop switch and it is not used when the firmware was set to use a BLTouch.

**Reply by cmacnichol** [reply]
Sorry, Should of mentioned that.  The filament sensor was one of the first things I checked and I have verified it is working as desired.  The same behavior is seen with it disabled as well

So just to confirm for my own knowledge, does it matter if the z_min is in the triggered state?  I knew it was not used, it just struck me as odd to see it triggered when nothing was plugged into that port.

Is there anything else I can do to confirm what is causing the pause and park behavior?  What other items could be causing it if it is not the Filament sensor or the z_min?

**Reply by mriscoc** [reply]
The Octoprint terminal should give you the reason of the pause.

**Reply by cmacnichol** [reply]
I am not seeing one looking through the terminal results, but maybe I am just missing something.  I disabled the Filament sensor to be sure, then replicated the issue twice.

```log
Changing monitoring state from "Offline" to "Opening serial connection"
Connecting to port /dev/ttyACM0, baudrate 115200
Changing monitoring state from "Opening serial connection" to "Connecting"
Connected to: Serial<id=0x688d7808, open=True>(port='/dev/ttyACM0', baudrate=115200, bytesize=8, parity='N', stopbits=1, timeout=10.0, xonxoff=False, rtscts=False, dsrdtr=False), starting monitor
Send: N0 M110 N0*125
Recv: ok N0 P15 B15
Send: N0 M110 N0*125
Changing monitoring state from "Connecting" to "Operational"
Recv: ok N0 P15 B15
Send: N0 M110 N0*125
Recv: ok N0 P15 B15
Send: N1 M115*39
Recv: FIRMWARE_NAME:Marlin 2.1.3 MRiscoC Ender3V2-SKRME3V3-BLTUBL-MPC, based on bugfix-2.1.x (Mar 12 2023 17:25:26) SOURCE_CODE_URL:github.com/mriscoc/Ender3V2S1 PROTOCOL_VERSION:1.0 MACHINE_TYPE:Ender3V2-SKRME3V3-BLTUBL-MPC EXTRUDER_COUNT:1 UUID:cede2a2f-41a2-4748-9b12-c55c62f367ff
Recv: Cap:SERIAL_XON_XOFF:0
Recv: Cap:BINARY_FILE_TRANSFER:1
Recv: Cap:EEPROM:1
Recv: Cap:VOLUMETRIC:1
Recv: Cap:AUTOREPORT_POS:0
Recv: Cap:AUTOREPORT_TEMP:1
Recv: Cap:PROGRESS:0
Recv: Cap:PRINT_JOB:1
Recv: Cap:AUTOLEVEL:1
Recv: Cap:RUNOUT:1
Recv: Cap:Z_PROBE:1
Recv: Cap:LEVELING_DATA:1
Recv: Cap:BUILD_PERCENT:1
Recv: Cap:SOFTWARE_POWER:0
Recv: Cap:TOGGLE_LIGHTS:0
Recv: Cap:CASE_LIGHT_BRIGHTNESS:0
Recv: Cap:EMERGENCY_PARSER:1
Recv: Cap:HOST_ACTION_COMMANDS:1
Recv: Cap:PROMPT_SUPPORT:1
Recv: Cap:SDCARD:1
Recv: Cap:MULTI_VOLUME:0
Recv: Cap:REPEAT:0
Recv: Cap:SD_WRITE:1
Recv: Cap:AUTOREPORT_SD_STATUS:0
Recv: Cap:LONG_FILENAME:1
Recv: Cap:LFN_WRITE:1
Recv: Cap:CUSTOM_FIRMWARE_UPLOAD:0
Recv: Cap:EXTENDED_M20:1
Recv: Cap:THERMAL_PROTECTION:1
Recv: Cap:MOTION_MODES:0
Recv: Cap:ARCS:1
Recv: Cap:BABYSTEPPING:1
Recv: Cap:CHAMBER_TEMPERATURE:0
Recv: Cap:COOLER_TEMPERATURE:0
Recv: Cap:MEATPACK:0
Recv: Cap:CONFIG_EXPORT:0
Recv: area:{full:{min:{x:0.00,y:0.00,z:0.00},max:{x:248.00,y:231.00,z:250.00}},work:{min:{x:0.00,y:0.00,z:0.00},max:{x:230.00,y:230.00,z:250.00}}}
Recv: ok N1 P15 B15
Recv:  T:145.79 /0.00 B:22.92 /0.00 @:0 B@:0
Send: M21
Recv: echo:SD card ok
Recv: ok P15 B15
Send: M117 Octoprint is connected
Recv: ok P15 B15
Send: M155 S2
Recv: ok P15 B15
Send: M876 P1
Recv: ok P15 B15
Send: M20 L
Recv: Begin file list
Recv: ENDER_~1.GCO 984 Ender_3_V2_bed_level.gcode
Recv: End file list
Recv: ok P15 B15
Recv:  T:144.73 /0.00 B:22.73 /0.00 @:0 B@:0
Recv:  T:143.85 /0.00 B:22.69 /0.00 @:0 B@:0
Recv:  T:142.99 /0.00 B:23.11 /0.00 @:0 B@:0
Recv:  T:142.06 /0.00 B:22.91 /0.00 @:0 B@:0
Recv:  T:141.15 /0.00 B:22.90 /0.00 @:0 B@:0
Recv:  T:140.30 /0.00 B:22.91 /0.00 @:0 B@:0
Recv:  T:139.35 /0.00 B:22.96 /0.00 @:0 B@:0
Recv:  T:138.53 /0.00 B:22.58 /0.00 @:0 B@:0
Recv:  T:137.56 /0.00 B:22.94 /0.00 @:0 B@:0
Recv:  T:136.73 /0.00 B:22.68 /0.00 @:0 B@:0
Recv:  T:135.80 /0.00 B:22.71 /0.00 @:0 B@:0
Recv:  T:134.93 /0.00 B:22.51 /0.00 @:0 B@:0
Recv:  T:134.11 /0.00 B:22.80 /0.00 @:0 B@:0
Changing monitoring state from "Operational" to "Starting"
Send: N0 M110 N0*125
Recv: ok N0 P15 B15
Send: N1 M75 Ender_3_V2_bed_level.gcode*110
Recv: ok N1 P15 B15
Changing monitoring state from "Starting" to "Printing"
Send: N2 G90*18
Recv: ok N2 P15 B15
Send: N3 M73 P0*84
Recv: ok N3 P15 B15
Send: N4 G28*23
Recv:  T:133.28 /0.00 B:22.65 /0.00 @:0 B@:0
Recv: echo:busy: processing
Printer seems to support the busy protocol, will adjust timeouts and set busy interval accordingly
Recv:  T:132.48 /0.00 B:22.70 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv:  T:131.69 /0.00 B:22.48 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv:  T:130.72 /0.00 B:22.65 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv:  T:129.87 /0.00 B:22.71 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv:  T:128.95 /0.00 B:22.67 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv:  T:128.12 /0.00 B:22.90 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv:  T:127.10 /0.00 B:23.04 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv:  T:126.24 /0.00 B:22.76 /0.00 @:0 B@:0
Recv: X:156.50 Y:122.00 Z:10.47 E:0.00 Count X:12520 Y:9760 Z:4188
Recv: ok N4 P15 B15
Send: N5 M73 P0*82
Recv: ok N5 P15 B15
Send: N6 M113 S2*103
Recv: ok N6 P15 B15
Send: N7 G1 Z5*96
Recv: ok N7 P14 B15
Send: N8 G1 X32 Y36*37
Recv: ok N8 P13 B15
Send: N9 G1 Z0*107
Recv: ok N9 P12 B15
Changing monitoring state from "Printing" to "Pausing"
Send: N10 M400*22
Recv:  T:125.21 /0.00 B:22.97 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv:  T:124.34 /0.00 B:22.71 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv:  T:123.37 /0.00 B:22.74 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv: ok N10 P15 B15
Send: N11 M25*36
Recv: //action:paused
Printer signalled that it paused, switching state...
Recv: //action:prompt_end
Recv: //action:prompt_begin Pause
Recv: //action:prompt_button Dismiss
Recv: //action:prompt_show
Recv:  T:122.60 /0.00 B:22.78 /0.00 @:0 B@:0
Send: N12 M876 S0*90
Recv: echo:busy: processing
Recv:  T:121.63 /0.00 B:22.95 /0.00 @:0 B@:0
Did not receive parseable position data from printer within 10.0s, continuing without it
Recv: echo:busy: processing
Recv:  T:120.75 /0.00 B:22.82 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv:  T:119.82 /0.00 B:22.83 /0.00 @:0 B@:0
Recv: X:240.00 Y:221.00 Z:20.00 E:0.00 Count X:19200 Y:17680 Z:8000
Recv: echo:Send M108 to resume
Recv: //action:prompt_end
Recv: //action:prompt_begin Nozzle Parked
Recv: //action:prompt_button Continue
Recv: //action:prompt_show
Recv: echo:busy: paused for user
Recv:  T:119.00 /0.00 B:22.74 /0.00 @:0 B@:0
Send: N13 M876 S0*91
Recv: echo:busy: processing
Recv:  T:118.01 /180.00 B:22.75 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:117.48 /180.00 B:22.78 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:117.97 /180.00 B:22.81 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:119.47 /180.00 B:22.77 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:122.06 /180.00 B:23.06 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:125.04 /180.00 B:22.71 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:128.76 /180.00 B:23.17 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:132.38 /180.00 B:22.77 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:136.45 /180.00 B:23.01 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:140.43 /180.00 B:22.85 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:144.61 /180.00 B:22.91 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:148.48 /180.00 B:22.78 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:152.74 /180.00 B:22.86 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:156.64 /180.00 B:22.74 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:160.81 /180.00 B:22.99 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:164.65 /180.00 B:22.76 /0.00 @:127 B@:0
Recv: echo:busy: processing
Recv:  T:168.77 /180.00 B:22.81 /0.00 @:102 B@:0
Recv: echo:busy: processing
Recv:  T:172.37 /180.00 B:22.96 /0.00 @:58 B@:0
Recv: echo:busy: processing
Recv:  T:176.06 /180.00 B:22.79 /0.00 @:31 B@:0
Recv: echo:busy: processing
Recv:  T:178.34 /180.00 B:22.71 /0.00 @:23 B@:0
Recv: //action:prompt_end
Recv: //action:prompt_begin Paused
Recv: //action:prompt_button PurgeMore
Recv: //action:prompt_button Continue
Recv: //action:prompt_show
Recv: echo:busy: paused for user
Recv:  T:180.26 /180.00 B:22.78 /0.00 @:17 B@:0
Recv: echo:busy: paused for user
Recv:  T:181.12 /180.00 B:22.85 /0.00 @:18 B@:0
Recv: //action:prompt_end
Recv: echo:busy: processing
Recv:  T:181.64 /180.00 B:22.71 /0.00 @:20 B@:0
Recv: echo:busy: processing
Recv:  T:181.52 /180.00 B:23.00 /0.00 @:25 B@:0
Recv: echo:busy: processing
Recv:  T:181.14 /180.00 B:22.73 /0.00 @:31 B@:0
Recv: echo:busy: processing
Recv:  T:180.68 /180.00 B:22.62 /0.00 @:37 B@:0
Recv: //action:resumed
Printer signalled that it resumed, switching state...
Changing monitoring state from "Pausing" to "Resuming"
Recv: ok N11 P15 B13
Recv: ok N12 P15 B14
Recv: ok N13 P15 B15
Send: N14 M73 P15*86
Recv: ok N14 P15 B15
Send: N15 M114*19
Recv: X:32.00 Y:36.00 Z:0.00 E:0.00 Count X:2560 Y:2880 Z:0
Recv: ok N15 P15 B15
Send: N16 M76*37
Recv: ok N16 P15 B15
Recv: //action:paused
Printer signalled that it paused, switching state...
Changing monitoring state from "Resuming" to "Pausing"
Send: N17 M117 Print was paused*16
Recv:  T:180.19 /180.00 B:22.65 /0.00 @:41 B@:0
Recv: echo:busy: processing
Recv:  T:179.89 /180.00 B:22.70 /0.00 @:43 B@:0
Recv: echo:busy: processing
Recv:  T:179.55 /180.00 B:22.86 /0.00 @:45 B@:0
Recv: echo:busy: processing
Recv:  T:179.23 /180.00 B:23.00 /0.00 @:46 B@:0
Recv: X:240.00 Y:221.00 Z:20.00 E:-2.00 Count X:19200 Y:17680 Z:8000
Recv: echo:Send M108 to resume
Recv: //action:prompt_end
Recv: //action:prompt_begin Nozzle Parked
Recv: //action:prompt_button Continue
Recv: //action:prompt_show
Recv: echo:busy: paused for user
Recv:  T:179.14 /180.00 B:22.92 /0.00 @:46 B@:0
Send: N18 M876 S0*80
Recv: echo:busy: processing
Recv: //action:prompt_end
Recv: //action:prompt_begin Paused
Recv: //action:prompt_button PurgeMore
Recv: //action:prompt_button Continue
Recv: //action:prompt_show
Recv:  T:179.19 /180.00 B:22.79 /0.00 @:45 B@:0
Send: N19 M876 S1*80
Recv: //action:prompt_end
Recv:  T:179.20 /180.00 B:22.68 /0.00 @:45 B@:0 W:?
Recv: echo:busy: processing
Recv:  T:179.21 /180.00 B:22.66 /0.00 @:44 B@:0 W:12
Recv:  T:179.18 /180.00 B:22.75 /0.00 @:45 B@:0
```

**Comment by Crazy-Charles** [comment]
Looking at that log it's pausing because your nozzle temperature is reporting its only at 125C which is below the minimum extrusion temperature and resumes at 180C (which is probably what the minimum extrusion temp is set at in the firmware)

**Reply by cmacnichol** [reply]
That is the confusing part.  The first pause is correct, due to temp.  But the second and third pause have no reason stated.  It shows the PurgeMore\Continue screen, but the filament sensor is disabled. So I cannot tell why it is pausing.

**Reply by Crazy-Charles** [reply]
On the others, the reported temp has dropped below 180 e.g. 179.18 with a set point of 180 - which does seem to be low, PLA+ i print at 215C, ASA at 245C, normal PLA i use 200C, try upping the set point and set your bed temp to 60C at the same time.

**Reply by mriscoc** [reply]
About your Z_MIN status, I don't have a SKR mini to test if that is the default value for this port (for 4.2.2 board, it is open) but I don't remember any effect into the printer if this port is TRIGGERED.

**Reply by cmacnichol** [reply]
Was hoping it was going to be that simple.  Upped the set point to 200\60, preheated, verified Filament sensor was disabled and then started a print.  Same behavior, It homes to Z Axis correctly, then goes to the start position.  As soon as it does that it pauses and moves to the Park position with a LCD Message prompting for PurgeMore or Continue.

As far as the normal state of the z-stop, it should be OPEN as well from what I understand.  I just do not know why it shows otherwise when the port is empty.  That is why I kept thinking it was the problem.  But if the firmware ignores that state when a BLTouch is configured, then it does not sound like that is the issue.

Any other idea on what I can test or rule out?  Will be trying the BTT Firmware tonight just to rule a hardware issue out.  Not sure if I want that to fix it or not as I really don't want to use that firmware.

**Reply by Crazy-Charles** [reply]
Not having the SKR board in my collection i dont know what may be wrong, however i would check all the connections to the board just to make sure the right component is plugged into the correct socket.
I will add that on my "current" controller board which is modified and has a replacement processor but based on the Creality 4.2.2. circuit, my Z axis end stop port always reports "triggered" but it doesn't affect anything, i could invert the logic in the source and make it show open but as i said, it all works 100% the way it is.

**Reply by mriscoc** [reply]
Thanks @Crazy-Charles for the confirmation about Z_MIN.
@cmacnichol are you using a precompiled version or a custom version? Did you try to print from SD?

**Reply by cmacnichol** [reply]
@mriscoc 
Precompiled.  Ender3V2-SKRME3V3-BLTUBL-MPC-20230312.bin

Same behavior from Octoprint and the SD Card.

Thought about compiling my own, but still can't point at one thing that is causing the behavior that it would fix.

**Reply by mriscoc** [reply]
Please, post a picture of the end-stop diagnostic tool.
Set in your start G-code script: `M412 S0 H0`

**Comment by cmacnichol** [comment, marked as answer]
So, this is interesting.  The M412 disable command made no difference but I realized as I was testing that I had only attempted to print my leveling print this entire time, which has worked before.  Changing the print to a small item and it started to print without issues.  I had to update the Z-Offset value as the print was way to high, but it was working.  I have inspected the GCode of the leveling print and nowhere in there does it have anything that should trigger this behavior.  The only large difference is that the leveling code runs a straight line around the edge of the bed while the small print is all in the center.  The printer is now working with a few minor leveling issues.

At this point I can confirm that the filament sensor was completely unrelated, so the purge message it was triggering was really confusing.  

@mriscoc @Crazy-Charles Thank you both very much for the help.  Still confused as to the the root cause, Maybe the offset?  But you got me pointed in the right direction, which was more than I was getting myself.
